hypertrace-gradle-docker-java-application-plugin/src/main/java/org/hypertrace/gradle/docker/HypertraceDockerJavaApplicationPlugin.java

```diff
@@ -66,12 +66,17 @@ private void addApplicationExtension(Project project) {
         .create(EXTENSION_NAME, HypertraceDockerJavaApplication.class, project.getName());
   }
 
-  private void updateDefaultJvmArgs(Project project) {
+  private void updateDefaultJvmArgs(Project project, HypertraceDockerJavaApplication javaApplication) {
     // by default allow reflective access for monitoring executor service
     // https://github.com/micrometer-metrics/micrometer/issues/2317#issuecomment-952700482
+    List<String> jvmArgs = new ArrayList<>(List.of("--add-opens", "java.base/java.util.concurrent=ALL-UNNAMED"));
+    // add generational zgc if it is java 21 base image
+    if (javaApplication.baseImage.getOrElse("").endsWith("java:21")) {
```

Comment:
nit - can we add better support for this? Rather than have services specify custom base images that we then try to match here, can we allow specifying (or perhaps inferring) the base image via the plugin?

Also in general, this is unlikely to work - the apply statement of a plugin where we're calling this function typically runs before a project's custom code so if a project were changing the base image, it would happen after we evaluate this. Instead, we want to keep things lazy